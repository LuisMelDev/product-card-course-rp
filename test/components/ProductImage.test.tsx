import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';

describe('it', () => {
  it('should render correcly', () => {
    const wrapper = renderer.create(
      <ProductCard
        product={{
          id: '1',
          title: 'Product 1',
        }}
      >
        {() => (
          <ProductImage img="https://pps.whatsapp.net/v/t61.24694-24/236690053_424819545992935_4236778729403962088_n.jpg?stp=dst-jpg_s96x96&ccb=11-4&oh=01_AVym6SKAoAn50w0yvVlWhnq_fj6DNkHLbKQ2B9RZQyS6rQ&oe=62C0B48B" />
        )}
      </ProductCard>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correcly without image', () => {
    const wrapper = renderer.create(
      <ProductCard
        product={{
          id: '1',
          title: 'Product 1',
          img: 'https://pps.whatsapp.net',
        }}
      >
        {() => <ProductImage />}
      </ProductCard>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
