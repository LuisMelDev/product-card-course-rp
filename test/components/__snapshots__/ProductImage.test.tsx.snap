// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correcly 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="https://pps.whatsapp.net/v/t61.24694-24/236690053_424819545992935_4236778729403962088_n.jpg?stp=dst-jpg_s96x96&ccb=11-4&oh=01_AVym6SKAoAn50w0yvVlWhnq_fj6DNkHLbKQ2B9RZQyS6rQ&oe=62C0B48B"
  />
</div>
`;

exports[`it should render correcly without image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="https://pps.whatsapp.net"
  />
</div>
`;
